### PR TITLE
[INTERNAL] couplings protocol refactoring

### PR DESCRIPTION
**Pull request by b-schubert**
Refactoring of the couplings protocol to reduce code redundancy.
I tested the refactored code on a small set of proteins and one complex.

Fixes issue #57

### CODE REVIEW

File: evcouplings/couplings/protocol.py [hunk 1]
```diff
@@ -294,63 +295,13 @@ def standard(**kwargs):
...
-            # create JSON output and write to file
-            f.write(
-                evzoom_json(c) + "\n"
-            )
-
-    # dump output config to YAML file for debugging/logging
-    write_config_file(prefix + ".couplings_standard.outcfg", outcfg)
-
+    is_single_segment = segments is None or len(segments) == 1
+    _postprocess_inference(ecs, kwargs, model, outcfg, prefix,
```
Comment (thomashopf) [inline evcouplings/couplings/protocol.py]: 1) please adjust line wrapping like this for consistency with rest of code:
_postprocess_inference(
    ecs, kwargs, model, outcfg, prefix,
    generate_enrichment=is_single_segment,
    generate_line_plot=is_single_segment
)

2) _postprocess_inference shouldn't have side effects besides writing files - especially not on outcfg
I really like the new dictionary unpacking syntax, so in the end this looks like this like in other protocols:
```
outcfg = {
    **outcfg,
   **_postprocess_inference(...),
}
```

File: evcouplings/couplings/protocol.py [hunk 2]
```diff
@@ -495,11 +428,11 @@ def complex(**kwargs):
         )
     )
 
-    ec_lines_pymol_script(
-        ecs_longrange.iloc[:L, :],
-        outcfg["ec_lines_pml_file"],
-        chain=chain_mapping
-    )
+    _postprocess_inference(ecs, kwargs, model, outcfg, prefix,
```
Comment (thomashopf) [inline evcouplings/couplings/protocol.py]: same as above

File: evcouplings/couplings/protocol.py [hunk 3]
```diff
@@ -679,30 +597,92 @@ def mean_field(**kwargs):
         ascending=False
     )
 
+    is_single_segment = segments is None or len(segments) == 1
+    _postprocess_inference(ecs, kwargs, model, outcfg, prefix,
```
Comment (thomashopf) [inline evcouplings/couplings/protocol.py]: same as above

File: evcouplings/couplings/protocol.py (hunk 1 above)
Comment (b-schubert) [reply evcouplings/couplings/protocol.py]: Made the requested changes.

**Comment by thomashopf** [comment]
Thanks!

**Pull request merged**